AsyncDisplayKit/TextKit/ASTextKitContext.mm

```diff
@@ -58,6 +69,21 @@ - (instancetype)initWithAttributedString:(NSAttributedString *)attributedString
   return self;
 }
 
+#pragma mark - Text Storage
+
+- (void)resetTextStorage
+{
+  std::lock_guard<std::mutex> l(_textKitMutex);
+  [self _resetTextStorage];
+}
+
+- (void)_resetTextStorage
+{
+  [_textStorage setAttributedString:_attributedString];
```

Comment:
At this level, it is unclear why _textStorage would ever change to a different string...in other words, it is very strange / not obvious that the truncater may externally modify the text storage but not set it back.  We'd want to comment about this at the least, but I think the fix should probably go in the truncater instead.